Stack/Opc.Ua.Core/Security/Certificates/CertificateIdentifier.cs

```diff
@@ -487,8 +503,187 @@ public ICertificateStore OpenStore()
+                    }
+                    break;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Map a security policy to a list of supported certificate types.
+        /// </summary>
+        /// <param name="securityPolicy"></param>
+        public static IList<NodeId> MapSecurityPolicyToCertificateTypes(string securityPolicy)
+        {
+            var result = new List<NodeId>();
+            switch (securityPolicy)
+            {
+                case SecurityPolicies.Basic128Rsa15:
+                case SecurityPolicies.Basic256:
+                    result.Add(ObjectTypeIds.RsaMinApplicationCertificateType);
+                    goto case SecurityPolicies.Basic256Sha256;
+                case SecurityPolicies.Basic256Sha256:
+                case SecurityPolicies.Aes128_Sha256_RsaOaep:
+                case SecurityPolicies.Aes256_Sha256_RsaPss:
+                    result.Add(ObjectTypeIds.RsaSha256ApplicationCertificateType);
+                    break;
+                case SecurityPolicies.ECC_nistP256:
+                    result.Add(ObjectTypeIds.EccNistP256ApplicationCertificateType);
+                    goto case SecurityPolicies.ECC_nistP384;
+                case SecurityPolicies.ECC_nistP384:
+                    result.Add(ObjectTypeIds.EccNistP384ApplicationCertificateType);
+                    break;
+                case SecurityPolicies.ECC_brainpoolP256r1:
+                    result.Add(ObjectTypeIds.EccBrainpoolP256r1ApplicationCertificateType);
+                    goto case SecurityPolicies.ECC_brainpoolP384r1;
+                case SecurityPolicies.ECC_brainpoolP384r1:
+                    result.Add(ObjectTypeIds.EccBrainpoolP384r1ApplicationCertificateType);
+                    break;
+                case SecurityPolicies.ECC_curve25519:
+                    result.Add(ObjectTypeIds.EccCurve25519ApplicationCertificateType);
+                    break;
+                case SecurityPolicies.ECC_curve448:
+                    result.Add(ObjectTypeIds.EccCurve448ApplicationCertificateType);
+                    break;
+                case SecurityPolicies.Https:
+                    result.Add(ObjectTypeIds.HttpsCertificateType);
+                    break;
+                default:
+                    break;
+            }
+            return result;
+        }
+
+
+        /// <summary>
+        /// Disposes and deletes the reference to the certificate.
+        /// </summary>
+        public void DisposeCertificate()
+        {
+            var certificate = m_certificate;
+            m_certificate = null;
+            Utils.SilentDispose(certificate);
+        }
         #endregion
 
+        #region Private Members
+        /// <summary>
+        /// The tags of the supported certificate types.
+        /// </summary>
+        private static Dictionary<uint, string> m_supportedCertificateTypes = new Dictionary<uint, string>() {
```

Comment:
isn`t this mssing here: 
            { ObjectTypes.EccApplicationCertificateType, "??"},